diff -Naur xz-utils-5.6.2/debian/changelog xz-utils-5.8.1/debian/changelog
--- xz-utils-5.6.2/debian/changelog	2024-07-03 21:51:43
+++ xz-utils-5.8.1/debian/changelog	2025-04-03 23:02:58
@@ -1,3 +1,30 @@
+xz-utils (5.8.1-1) unstable; urgency=medium
+
+  * Import 5.8.1
+    - Threaded .xz decoder frees memory too early (CVE-2025-31115)
+  * Upload to unstable
+
+ -- Sebastian Andrzej Siewior <sebastian@breakpoint.cc>  Thu, 03 Apr 2025 23:02:58 +0200
+
+xz-utils (5.8.0-1) experimental; urgency=medium
+
+  * Import 5.8.0
+
+ -- Sebastian Andrzej Siewior <sebastian@breakpoint.cc>  Wed, 26 Mar 2025 23:01:15 +0100
+
+xz-utils (5.6.4-1) unstable; urgency=medium
+
+  * Import 5.6.4
+  * Sync output file before removing source file (Closes: #814089).
+
+ -- Sebastian Andrzej Siewior <sebastian@breakpoint.cc>  Sat, 01 Mar 2025 21:52:57 +0100
+
+xz-utils (5.6.3-1) unstable; urgency=medium
+
+  * Import 5.6.3
+
+ -- Sebastian Andrzej Siewior <sebastian@breakpoint.cc>  Thu, 24 Oct 2024 21:27:12 +0200
+
 xz-utils (5.6.2-2) unstable; urgency=medium
 
   * Increase symbol dependency on 5.6.0 symbols to 5.6.2 due to the revert.
diff -Naur xz-utils-5.6.2/debian/control xz-utils-5.8.1/debian/control
--- xz-utils-5.6.2/debian/control	2024-07-03 21:39:45
+++ xz-utils-5.8.1/debian/control	2025-04-03 22:58:15
@@ -8,7 +8,7 @@
  gettext, autopoint | gettext (<< 0.18-1), autopoint | cvs, po4a
 Build-Depends-Indep: doxygen, graphviz
 Build-Conflicts: automake1.4
-Standards-Version: 4.6.2
+Standards-Version: 4.7.2
 Vcs-Browser: https://salsa.debian.org/debian/xz-utils
 Vcs-Git: https://salsa.debian.org/debian/xz-utils.git
 Homepage: https://tukaani.org/xz/
diff -Naur xz-utils-5.6.2/debian/gbp.conf xz-utils-5.8.1/debian/gbp.conf
--- xz-utils-5.6.2/debian/gbp.conf	2024-06-13 22:42:06
+++ xz-utils-5.8.1/debian/gbp.conf	2025-04-03 22:58:15
@@ -1,6 +1,6 @@
 [DEFAULT]
 debian-branch=debian/unstable
-upstream-branch=upstream/v5.6
+upstream-branch=upstream/v5.8
 
 [pq]
 patch-numbers = False
diff -Naur xz-utils-5.6.2/debian/symbols xz-utils-5.8.1/debian/symbols
--- xz-utils-5.6.2/debian/symbols	2024-07-02 23:17:14
+++ xz-utils-5.8.1/debian/symbols	2025-04-03 22:58:53
@@ -24,5 +24,5 @@
  (arch=linux-any)lzma_stream_encoder_mt@XZ_5.2.2 5.2.2
  (arch=linux-any)lzma_stream_encoder_mt_memusage@XZ_5.1.2alpha 5.4
  (arch=linux-any)lzma_stream_encoder_mt_memusage@XZ_5.2.2 5.2.2
- XZ_5.6.0@XZ_5.6.0 5.6.2
- lzma_mt_block_size@XZ_5.6.0 5.6.2
+ *@XZ_5.6.0 5.6.2
+ *@XZ_5.8 5.8.0
diff -Naur xz-utils-5.6.2/debian/tests/testsuite xz-utils-5.8.1/debian/tests/testsuite
--- xz-utils-5.6.2/debian/tests/testsuite	2024-06-13 22:41:58
+++ xz-utils-5.8.1/debian/tests/testsuite	2025-04-03 22:58:15
@@ -2,7 +2,7 @@
 
 # Tweak the Makefile to link against the installed liblzma for the tests
 sed 's@..top_builddir..src.liblzma.liblzma.la@-llzma@' -i tests/Makefile.am
-
+./autogen.sh
 ./configure
 
 exec make check
diff -Naur xz-utils-5.6.2/debian/upstream/signing-key.asc xz-utils-5.8.1/debian/upstream/signing-key.asc
--- xz-utils-5.6.2/debian/upstream/signing-key.asc	2024-06-13 22:41:58
+++ xz-utils-5.8.1/debian/upstream/signing-key.asc	2025-04-03 22:58:15
@@ -13,18 +13,18 @@
 MtbltZo0ysPMa6Xj5xFaYqWk/BI4iLb2Gs+ByGo/+a0Eq4XYBMOpitNniQARAQAB
 tCdMYXNzZSBDb2xsaW4gPGxhc3NlLmNvbGxpbkB0dWthYW5pLm9yZz6JAlEEEwEK
 ADsCGwMCHgECF4AECwkIBwMVCggFFgIDAQAWIQQ2kMJAzlG0Zw0wrRw47nV9aRhG
-IAUCZZwJyQUJGuHiNwAKCRA47nV9aRhGIE4qD/4jdFTe3WPpLgvz/jdlbnSZxr7q
-OS6H/ZJFENHO4SbavXdoXLtj+t6/lqWq890Js8IpWaaiJLowzW1xJMEg99W6k0KD
-3pHUbwPxf0GCSAt/W4JYxdTj+1ggdHjx5yBAmOakjnOH+ZDKQNBnDOI6ghf3ew+H
-9z/b0mQX3rlQbtoqSPZtuDOdFcjCOSwEyqdV+9eNqnv2CoKZkiGoUB1WGCbqKUkY
-KiUJ3WldmPQ5RQYjEi7zZWVac1VuwBA0XOku+W4cCJ5DnPyK7CtMwC84VvaodlOX
-UAK3Y5BIZpZM2Rk6yMX5lFDA5nA8UuHJQRDjTVmh3BIdgRvp0ZV6ogtqNE7RifpW
-aBWDIsCkimcbCJJM+edOLiVZog+ia1Ts8zu33wj7Tnvp5znLc8NLZIqwu1HKLS97
-m+Yf5oC3ObTZtXbVF+OglWe/3ljLHdL2bJxNdtcVlChSNPUW3fgLHk9Fzrlnqdab
-tSGwI/0Ryt00cKjRiMOagTn5Nly6boCtgGYdQafQoSrs3eQjnWVgbNYDMgPyl4k+
-Q5RJLEY7AvtXo7FUEgOTfr9PWmjmc2JzGpxbtwl6sQi6yLrBZTRf1Xao2OjOje6G
-XdUbXNmgOv16sWxcI0s4lX1z28BgHQfwXhBFBRjw2Sy+6TfFXjX24thcpMwvyJ3c
-xhMtdY4N4jyfRjYe8LkCDQRMxDmSARAAv8XAp2PGA/G1KmCrVIzOBm1NPIuqGAYP
+IAUCZ364UgUJHMSQwAAKCRA47nV9aRhGIH6dD/46yxNWPrEQlVFd2XbOfBdYjVSh
+SOtnF/7nfdpxObjsy5eqsuNGGknkfJkqDvYXcvtVLwn+U3WEscCX3eZ/syM+bpMj
++rMUfczDR21z6UEG9nfR5Yx+7TOpPfXylr2W1OfAtjdMv5UxYKpmmaQogVH+EeA0
+zAE0gZ+n1h0CzAXjmSeNOmAhaMcR1kkA7jM4ZW3r7ovcUuHSR6XwYl2gCDRkHaFR
+WePNaOUXRHn/WQFQusUZ8FgZdidw3j4U13ajpoQdZonXk9lBrxu3nS2EzaTuPsEf
+R9f1xRlWn733sTAJ5jCOpDFWvmLMNowodC5FYd3SKIn6K80uPamcVnPrtcfGZXAI
+OgELAmbrDRz3uDfK1sty4pqI7We/0l0poqf0i863piA0a+awOMURTo9B/vjJvdh1
+Rf672fVWfnWRfAnjgIifCN4nBPVKRcY9XgmOG8ZqQgpOug7EERz4W8IZccjxx4gu
+jTMXoPUY1s+6c/CJlQ3gESlfPkz1UYKCFjEvApMOD99SG8AUJvQbGvPFkIK5euWS
+hOVLAmLKI3YNGCXUiPA4i9AEYnkDtAydohUKU/WclBWHWaeAZDccDivs8dPiF3Dd
+YrR0eb81lB36Jm13cJodBS4aArHvgDeDvz7aLr7An0qXmNTnpc3cdU1ovskNXmA2
+Yj+I0ojpJsh6ici6dLkCDQRMxDmSARAAv8XAp2PGA/G1KmCrVIzOBm1NPIuqGAYP
 c1l9p0dYdhEgvfw0NXcl5MDv1jbOPZ2PspA8NP7Rqp6LNNXYTeM/eIJDndU5Phyi
 ewFpACAp7Gmm2dL5PUOhu0gIUnQYbN/QdGPoo7bNI646K1Y9aVTBu9fszQssjb6G
 qXHSNM+pskVn9lropO1tLrF0I9VSlSphlCmiQRlzBCZSnxD6UagkPaw1gJnJqnrd
@@ -36,17 +36,17 @@
 9jGyvwr3T37gfzYCNMoZf8GaxAUJMCoGTqnsjTPGMion/DfdNkFDQ+fivdYiVQ9p
 /Njpr38sC83V8dHF/1KkIHImyzMPTdC7l/lMHyC2Gx2dWZOjuOOKit0Qoy3DZoQw
 vN1ZZND9M1UAEQEAAYkCPAQYAQoAJgIbDBYhBDaQwkDOUbRnDTCtHDjudX1pGEYg
-BQJlnAmyBQka4eIgAAoJEDjudX1pGEYguyYQAJo+5SnMMdu+d70mWfUb9PZg7P5C
-GRepHnckx9Sis5oR5s7NNl5j5Yy4J1UwsmrP+mn52ujqewkkVsCq65NGQQx7+tkw
-uKGvnGBkHdrI+aJk86qLMf4DlnNJEmN8t5jTGQfRLbFVf2I8EY6qXAzCSmL9Zs++
-rDUz65GOTB1EP0XmBRsuVYRfDbFezrPQH0JDucbXFi/2BDnl2/Mk9NBoQ0CvB4oG
-tLDiQZ+jV7n1VXXJ1faD9s7i0hOTdcG6rlyIqi/LyAzdCnOYTkmv3U1kdmzkvrh1
-KEiejnM5fj27RE2v191vh3hgZ+X5+uwjNTP0QC4qP8XykQOAA8usOMVZ72lyXCAk
-wiUcRdrAXLN/XbIFNcQ3m4d3W6t60Gk09wFlUKaEltDMlPUsxiSG3qFwFGPBP6UV
-h3mjJMAl1jltLrR7ybez0SczfrcAtdCsKTvgzV9W2TzUfK2R9PBanmXTXK2M7yU3
-IquHt3Je4aSP7XYb5D+ajlbFNvnXOYcai8WryfC5nLAfV4MbPX+UlRaYCqqHVhut
-gK93re1L5mMI3zjG5Ri5jLpUA9toSJCIJIY5zwr/8LL/ZL4TixXlouA17yjkpY/e
-Bjs8cNj1O3aM4jY2FKCS8UbfxOiARk/5kBMRPEZ/mqpMQttzE8KVjOv6fRxy/eVE
-888/gToe5kb8qYwy
-=6rZC
+BQJnfrhHBQkcxJC1AAoJEDjudX1pGEYgjaIQAIxs2u0oNqKiFK58d/lpSg+sWgwO
+3WG9cChhQvPnJdyAdXKlnw8/SSykqtuMv1/N29mYSO7K3sCISvoCTR2KqT546o+C
+2/CqSs3WEuX1SpPQGVPULiTxLLsZh0WhJrKchmV3NRACPxoueiqtFosJPA6u1sf2
+ilN89ArGxzteIuTpyUy5r98GHUK5svjVh4sz+KvTNlGFHN3nAoR7BhiDf6RgQ0Q3
+ukkGJqvtYSUV3RuVkgR+7FHDFsb6RRmykcm+YStz3P0YTMJjbLYewaQLzKZnLWnS
+N18NRrITwEUicBXkcOFjF18d7jXYsAzzX7l28LFi5ygWuOOvT5s7EzTmWty5+qre
+HHnjL4SCQY2rkb9AipUTqpoDhBz91QivTdQ/EsJlP6e08hwaa7heZqZWWy1eRr8R
+EzD2r3GfQ2EegPkwsidivDBlK4PgWnlEl06RbE1dXoamXfPJ1q0SFlkw3zMW+bqY
+P1jTxrJVAlXFf8qT3d5pXz871R1NPP/dyPJni844Bi42q8JHIZaKWZ6QQsP20TGM
+BrvbI8qMCM1gH58vcBWBHtpm2kGEjVdJyl75RgjCf1VHqPab6A0/+tABGlLf5rm9
+kxOmb2wDCW8xPno/A8psVWcYE7gERT6Bdqbkr3SZw92izGCA4m17O1MPf6vj/ngb
+RROvuEI11HLOmBJv
+=m5Nx
 -----END PGP PUBLIC KEY BLOCK-----
diff -Naur xz-utils-5.6.2/debian/watch xz-utils-5.8.1/debian/watch
--- xz-utils-5.6.2/debian/watch	2024-06-13 22:41:58
+++ xz-utils-5.8.1/debian/watch	2025-04-03 22:58:15
@@ -1,4 +1,4 @@
 version=4
 opts=dversionmangle=s/\+\d{8}$//,pgpsigurlmangle=s/$/.sig/ \
 https://tukaani.org/xz/		\
-https://github.com/tukaani-project/xz/releases/download/v[^/]+/xz-([\d\.]*(?:beta|alpha)?)\.tar\.xz
+https://github.com/tukaani-project/xz/releases/download/v[^/]+/xz-(5\.8\.[\d\.]*(?:beta|alpha)?)\.tar\.xz
