diff -Naur xz-utils-5.6.0/debian/changelog xz-utils-5.8.1/debian/changelog
--- xz-utils-5.6.0/debian/changelog	2024-02-28 20:03:45
+++ xz-utils-5.8.1/debian/changelog	2025-04-03 23:02:58
@@ -1,3 +1,61 @@
+xz-utils (5.8.1-1) unstable; urgency=medium
+
+  * Import 5.8.1
+    - Threaded .xz decoder frees memory too early (CVE-2025-31115)
+  * Upload to unstable
+
+ -- Sebastian Andrzej Siewior <sebastian@breakpoint.cc>  Thu, 03 Apr 2025 23:02:58 +0200
+
+xz-utils (5.8.0-1) experimental; urgency=medium
+
+  * Import 5.8.0
+
+ -- Sebastian Andrzej Siewior <sebastian@breakpoint.cc>  Wed, 26 Mar 2025 23:01:15 +0100
+
+xz-utils (5.6.4-1) unstable; urgency=medium
+
+  * Import 5.6.4
+  * Sync output file before removing source file (Closes: #814089).
+
+ -- Sebastian Andrzej Siewior <sebastian@breakpoint.cc>  Sat, 01 Mar 2025 21:52:57 +0100
+
+xz-utils (5.6.3-1) unstable; urgency=medium
+
+  * Import 5.6.3
+
+ -- Sebastian Andrzej Siewior <sebastian@breakpoint.cc>  Thu, 24 Oct 2024 21:27:12 +0200
+
+xz-utils (5.6.2-2) unstable; urgency=medium
+
+  * Increase symbol dependency on 5.6.0 symbols to 5.6.2 due to the revert.
+    Thanks to Ben Hutchings (Closes: #1074602).
+  * Correct the `terse' build. Thanks to Ben Hutchings (Closes: #1074604).
+
+ -- Sebastian Andrzej Siewior <sebastian@breakpoint.cc>  Wed, 03 Jul 2024 21:51:43 +0200
+
+xz-utils (5.6.2-1) unstable; urgency=medium
+
+  * Import 5.6.2
+  * Takeover maintenance of the package.
+  * Revert the debian/ mostly to what was in 5.6.1-1
+
+ -- Sebastian Andrzej Siewior <sebastian@breakpoint.cc>  Sat, 15 Jun 2024 10:42:31 +0200
+
+xz-utils (5.6.1+really5.4.5-1) unstable; urgency=critical
+
+  * Non-maintainer upload by the Security Team.
+  * Revert back to the 5.4.5-0.2 version
+
+ -- Salvatore Bonaccorso <carnil@debian.org>  Thu, 28 Mar 2024 15:59:38 +0100
+
+xz-utils (5.6.1-1) unstable; urgency=medium
+
+  * Non-maintainer upload.
+  * Import 5.6.1 (Closes: #1067708).
+  * Takeover maintenance of the package.
+
+ -- Sebastian Andrzej Siewior <sebastian@breakpoint.cc>  Wed, 27 Mar 2024 22:53:21 +0100
+
 xz-utils (5.6.0-0.2) unstable; urgency=medium
 
   * Non-maintainer upload.
diff -Naur xz-utils-5.6.0/debian/control xz-utils-5.8.1/debian/control
--- xz-utils-5.6.0/debian/control	2024-02-26 07:47:42
+++ xz-utils-5.8.1/debian/control	2025-04-03 22:58:15
@@ -1,17 +1,17 @@
 Source: xz-utils
 Section: utils
 Priority: optional
-Maintainer: Jonathan Nieder <jrnieder@gmail.com>
-Uploaders: Mohammed Adnène Trojette <adn+deb@diwi.org>
+Maintainer: Sebastian Andrzej Siewior <sebastian@breakpoint.cc>
+Uploaders: Jonathan Nieder <jrnieder@gmail.com>
 Build-Depends: debhelper-compat (= 13), dpkg-dev (>= 1.16.2),
  autoconf (>= 2.64~), automake, libtool (>= 2.2),
  gettext, autopoint | gettext (<< 0.18-1), autopoint | cvs, po4a
-Build-Depends-Indep: doxygen
+Build-Depends-Indep: doxygen, graphviz
 Build-Conflicts: automake1.4
-Standards-Version: 4.6.2
+Standards-Version: 4.7.2
 Vcs-Browser: https://salsa.debian.org/debian/xz-utils
 Vcs-Git: https://salsa.debian.org/debian/xz-utils.git
-Homepage: https://xz.tukaani.org/xz-utils/
+Homepage: https://tukaani.org/xz/
 Rules-Requires-Root: no
 
 Package: liblzma5
diff -Naur xz-utils-5.6.0/debian/copyright xz-utils-5.8.1/debian/copyright
--- xz-utils-5.6.0/debian/copyright	2024-02-26 07:47:42
+++ xz-utils-5.8.1/debian/copyright	2025-04-03 22:58:15
@@ -1,10 +1,10 @@
 Format: https://www.debian.org/doc/packaging-manuals/copyright-format/1.0/
 Upstream-Name: XZ Utils
 Upstream-Contact:
- xz@tukaani.org
- https://xz.tukaani.org/contact.html
+ Lasse Collin <lasse.collin@tukaani.org>
+ https://tukaani.org/xz/lists.html
 Source:
- https://xz.tukaani.org/xz-utils
+ https://tukaani.org/xz
  https://github.com/tukaani-project/xz
 Comment:
  XZ Utils is developed and maintained upstream by Lasse Collin.  Major
@@ -17,8 +17,7 @@
  documentation come from Doxygen.
 License:
  Different licenses apply to different files in this package. Here
- is a rough summary of which licenses apply to which parts of this
- package (but check the individual files to be sure!):
+ is a summary of which licenses apply to which parts of this package:
  .
    - liblzma is under the BSD Zero Clause License (0BSD).
  .
@@ -39,20 +38,6 @@
      lzma-file-format.xt are in the public domain but may
      be distributed under the terms of 0BSD too.
  .
-   - Doxygen-generated HTML version of the liblzma API documentation:
-     While Doxygen is under the GNU GPLv2, the license information
-     in Doxygen includes the following exception:
- .
-         Documents produced by doxygen are derivative works
-         derived from the input used in their production;
-         they are not affected by this license.
- .
-     Note: The JavaScript files (under the MIT license) have
-     been removed from the Doxygen output.
- .
-   - The XZ logo (xz-logo.png) included in the Doxygen-generated
-     documentation is under the Creative Commons BY-SA 4.0 license.
- .
    - Translated messages and man pages are under 0BSD except that
      some old translations are in the public domain.
  .
@@ -65,8 +50,9 @@
      These files don't affect the licensing of the binaries being
      built.
  .
-   - The extra directory contain files that are under various
-     free software licenses.
+   - The 'extra' directory contains files that are under various
+     free software licenses. These aren't built or installed as
+     part of XZ Utils.
  .
  For the files under the BSD Zero Clause License (0BSD), if
  a copyright notice is needed, the following is sufficient:
@@ -79,24 +65,39 @@
  it is not legally required by the license terms. Here is an example
  of a good notice to put into "about box" or into documentation:
  .
-     This software includes code from XZ Utils
-     <https://xz.tukaani.org/xz-utils/>.
+     This software includes code from XZ Utils <https://tukaani.org/xz/>.
  .
  The following license texts are included in the following files:
    - COPYING.0BSD: BSD Zero Clause License
    - COPYING.LGPLv2.1: GNU Lesser General Public License version 2.1
    - COPYING.GPLv2: GNU General Public License version 2
    - COPYING.GPLv3: GNU General Public License version 3
-   - COPYING.CC-BY-SA-4.0: Creative Commons Attribution-ShareAlike 4.0
-                           International Public License
  .
+ A note about old XZ Utils releases:
+ .
+     XZ Utils releases 5.4.6 and older and 5.5.1alpha have a
+     significant amount of code put into the public domain and
+     that obviously remains so. The switch from public domain to
+     0BSD for newer releases was made in Febrary 2024 because
+     public domain has (real or perceived) legal ambiguities in
+     some jurisdictions.
+ .
+     There is very little *practical* difference between public
+     domain and 0BSD. The main difference likely is that one
+     shouldn't claim that 0BSD-licensed code is in the public
+     domain; 0BSD-licensed code is copyrighted but available under
+     an extremely permissive license. Neither 0BSD nor public domain
+     require retaining or reproducing author, copyright holder, or
+     license notices when distributing the software. (Compare to,
+     for example, BSD 2-Clause "Simplified" License which does have
+     such requirements.)
+ .
  If you have questions, don't hesitate to ask for more information.
  The contact information is in the README file.
 
 Files: *
 Copyright: Lasse Collin
 	   Igor Pavlov
-	   Jia Tan
 	   Andrew Dudman
 	   Chenxi Mao
 	   Ilya Kurdyukov
@@ -108,17 +109,6 @@
 	   Sebastian Andrzej Siewior
 	   Wei Dai
 License: 0BSD
- Permission to use, copy, modify, and/or distribute this
- software for any purpose with or without fee is hereby granted.
- .
- THE SOFTWARE IS PROVIDED "AS IS" AND THE AUTHOR DISCLAIMS ALL
- WARRANTIES WITH REGARD TO THIS SOFTWARE INCLUDING ALL IMPLIED
- WARRANTIES OF MERCHANTABILITY AND FITNESS. IN NO EVENT SHALL
- THE AUTHOR BE LIABLE FOR ANY SPECIAL, DIRECT, INDIRECT, OR
- CONSEQUENTIAL DAMAGES OR ANY DAMAGES WHATSOEVER RESULTING FROM
- LOSS OF USE, DATA OR PROFITS, WHETHER IN AN ACTION OF CONTRACT,
- NEGLIGENCE OR OTHER TORTIOUS ACTION, ARISING OUT OF OR IN
- CONNECTION WITH THE USE OR PERFORMANCE OF THIS SOFTWARE.
 
 Files: src/scripts/* lib/* extra/scanlzma/scanlzma.c
 Copyright: © 1993, Jean-loup Gailly
@@ -146,11 +136,7 @@
            © 2019 Marc Stevens <marc.stevens@cwi.nl>
 License: GPL-3.0-or-later-WITH-Autoconf-exception-macro
 
-Files: doc/xz-logo.png
-Copyright: © 2023 by Jia Tan
-License: CC-BY-SA-4.0
-
-Files: po4a/.gitignore po/LINGUAS
+Files: po/LINGUAS
 Copyright: none; these are just short lists.
 License: none
  No copyright to license.
@@ -215,6 +201,19 @@
  The Debian packaging files are in the public domain.
  You may freely use, modify, distribute, and relicense them.
 
+License: 0BSD
+ Permission to use, copy, modify, and/or distribute this
+ software for any purpose with or without fee is hereby granted.
+ .
+ THE SOFTWARE IS PROVIDED "AS IS" AND THE AUTHOR DISCLAIMS ALL
+ WARRANTIES WITH REGARD TO THIS SOFTWARE INCLUDING ALL IMPLIED
+ WARRANTIES OF MERCHANTABILITY AND FITNESS. IN NO EVENT SHALL
+ THE AUTHOR BE LIABLE FOR ANY SPECIAL, DIRECT, INDIRECT, OR
+ CONSEQUENTIAL DAMAGES OR ANY DAMAGES WHATSOEVER RESULTING FROM
+ LOSS OF USE, DATA OR PROFITS, WHETHER IN AN ACTION OF CONTRACT,
+ NEGLIGENCE OR OTHER TORTIOUS ACTION, ARISING OUT OF OR IN
+ CONNECTION WITH THE USE OR PERFORMANCE OF THIS SOFTWARE.
+
 License: LGPL-2.1+
  This program is free software; you can redistribute it and/or modify
  it under the terms of the GNU Lesser General Public License as published by
@@ -233,21 +232,6 @@
  On Debian systems, the complete text of the GNU Lesser General Public
  License version 2.1 can be found in ‘/usr/share/common-licenses/LGPL-2.1’.
 
-License: GPL-2
- Permission to use, copy, modify, and distribute this software and its
- documentation under the terms of the GNU General Public License is
- hereby granted. No representations are made about the suitability of
- this software for any purpose. It is provided "as is" without express
- or implied warranty. See the GNU General Public License for more
- details.
- .
- Documents produced by doxygen are derivative works derived from the
- input used in their production; they are not affected by this license.
- .
- On Debian systems, the complete text of the version of the GNU General
- Public License distributed with Doxygen can be found in
- ‘/usr/share/common-licenses/GPL-2’.
-
 License: GPL-2+
  This program is free software; you can redistribute it and/or modify
  it under the terms of the GNU General Public License as published by
@@ -266,7 +250,7 @@
  On Debian systems, the complete text of the GNU General Public License
  version 2 can be found in ‘/usr/share/common-licenses/GPL-2’.
 
-License: Autoconf
+License: GPL-3.0-or-later-WITH-Autoconf-exception-macro
  This program is free software: you can redistribute it and/or modify it
  under the terms of the GNU General Public License as published by the
  Free Software Foundation, either version 3 of the License, or (at your
@@ -306,9 +290,3 @@
  are permitted in any medium without royalty provided the copyright
  notice and this notice are preserved.  This file is offered as-is,
  without warranty of any kind.
-
-License: CC-BY-SA-4.0
- This work is licenced under the Creative Commons Attribution-Share Alike 4.0
- International License. To view a copy of this licence, visit
- https://creativecommons.org/licenses/by-sa/4.0/ or send a letter to Creative
- Commons, 171 Second Street, Suite 300, San Francisco, California 94105, USA.
diff -Naur xz-utils-5.6.0/debian/gbp.conf xz-utils-5.8.1/debian/gbp.conf
--- xz-utils-5.6.0/debian/gbp.conf	2024-02-26 07:48:08
+++ xz-utils-5.8.1/debian/gbp.conf	2025-04-03 22:58:15
@@ -1,6 +1,6 @@
 [DEFAULT]
 debian-branch=debian/unstable
-upstream-branch=upstream/v5.5
+upstream-branch=upstream/v5.8
 
 [pq]
 patch-numbers = False
diff -Naur xz-utils-5.6.0/debian/liblzma-doc.docs xz-utils-5.8.1/debian/liblzma-doc.docs
--- xz-utils-5.6.0/debian/liblzma-doc.docs	2024-01-31 21:32:41
+++ xz-utils-5.8.1/debian/liblzma-doc.docs	2025-04-03 22:58:15
@@ -1 +1 @@
-doc/api/
+usr/share/doc/xz/api
diff -Naur xz-utils-5.6.0/debian/patches/series xz-utils-5.8.1/debian/patches/series
--- xz-utils-5.6.0/debian/patches/series	2024-02-28 19:58:06
+++ xz-utils-5.8.1/debian/patches/series	1970-01-01 01:00:00
@@ -1,2 +0,0 @@
-xz-Add-missing-RISC-V-on-the-filter-list-in-the-man-page.patch
-xz-Change-logging-level-for-thread-reduction-to-highest-v.patch
diff -Naur xz-utils-5.6.0/debian/patches/xz-Add-missing-RISC-V-on-the-filter-list-in-the-man-page.patch xz-utils-5.8.1/debian/patches/xz-Add-missing-RISC-V-on-the-filter-list-in-the-man-page.patch
--- xz-utils-5.6.0/debian/patches/xz-Add-missing-RISC-V-on-the-filter-list-in-the-man-page.patch	2024-02-28 19:58:06
+++ xz-utils-5.8.1/debian/patches/xz-Add-missing-RISC-V-on-the-filter-list-in-the-man-page.patch	1970-01-01 01:00:00
@@ -1,31 +0,0 @@
-From: Chien Wong <m@xv97.com>
-Date: Sun, 25 Feb 2024 21:38:13 +0800
-Subject: xz: Add missing RISC-V on the filter list in the man page
-
-Signed-off-by: Chien Wong <m@xv97.com>
----
- src/xz/xz.1 | 4 +++-
- 1 file changed, 3 insertions(+), 1 deletion(-)
-
-diff --git a/src/xz/xz.1 b/src/xz/xz.1
-index 6c8a52f..6c34062 100644
---- a/src/xz/xz.1
-+++ b/src/xz/xz.1
-@@ -4,7 +4,7 @@
- .\" Authors: Lasse Collin
- .\"          Jia Tan
- .\"
--.TH XZ 1 "2024-02-13" "Tukaani" "XZ Utils"
-+.TH XZ 1 "2024-02-25" "Tukaani" "XZ Utils"
- .
- .SH NAME
- xz, unxz, xzcat, lzma, unlzma, lzcat \- Compress or decompress .xz and .lzma files
-@@ -1812,6 +1812,8 @@ and
- \fB\-\-ia64\fR[\fB=\fIoptions\fR]
- .TP
- \fB\-\-sparc\fR[\fB=\fIoptions\fR]
-+.TP
-+\fB\-\-riscv\fR[\fB=\fIoptions\fR]
- .PD
- Add a branch/call/jump (BCJ) filter to the filter chain.
- These filters can be used only as a non-last filter
diff -Naur xz-utils-5.6.0/debian/patches/xz-Change-logging-level-for-thread-reduction-to-highest-v.patch xz-utils-5.8.1/debian/patches/xz-Change-logging-level-for-thread-reduction-to-highest-v.patch
--- xz-utils-5.6.0/debian/patches/xz-Change-logging-level-for-thread-reduction-to-highest-v.patch	2024-02-28 19:58:06
+++ xz-utils-5.8.1/debian/patches/xz-Change-logging-level-for-thread-reduction-to-highest-v.patch	1970-01-01 01:00:00
@@ -1,45 +0,0 @@
-From: Jia Tan <jiat0218@gmail.com>
-Date: Tue, 27 Feb 2024 23:42:41 +0800
-Subject: xz: Change logging level for thread reduction to highest verbosity
- only.
-
-Now that multi threaded encoding is the default, users do not need to
-see a warning message everytime the number of threads is reduced. On
-some machines, this could happen very often. It is not unreasonable for
-users to need to set double verbose mode to see this kind of
-information.
-
-To see these warning messages -vv or --verbose --verbose must be passed
-to set xz into the highest possible verbosity mode.
-
-These warnings had caused automated testing frameworks to fail when they
-expected no output to stderr.
-
-Thanks to Sebastian Andrzej Siewior for reporting this and for the
-initial version of the patch.
----
- src/xz/coder.c | 4 ++--
- 1 file changed, 2 insertions(+), 2 deletions(-)
-
-diff --git a/src/xz/coder.c b/src/xz/coder.c
-index 4efaa80..a408173 100644
---- a/src/xz/coder.c
-+++ b/src/xz/coder.c
-@@ -581,7 +581,7 @@ coder_set_compression_settings(void)
- 
- 			if (memory_usage <= memory_limit) {
- 				// The memory usage is now low enough.
--				message(V_WARNING, _("Reduced the number of "
-+				message(V_DEBUG, _("Reduced the number of "
- 					"threads from %s to %s to not exceed "
- 					"the memory usage limit of %s MiB"),
- 					uint64_to_str(
-@@ -601,7 +601,7 @@ coder_set_compression_settings(void)
- 		// time the soft limit will never make xz fail and never make
- 		// xz change settings that would affect the compressed output.
- 		if (hardware_memlimit_mtenc_is_default()) {
--			message(V_WARNING, _("Reduced the number of threads "
-+			message(V_DEBUG, _("Reduced the number of threads "
- 				"from %s to one. The automatic memory usage "
- 				"limit of %s MiB is still being exceeded. "
- 				"%s MiB of memory is required. "
diff -Naur xz-utils-5.6.0/debian/rules xz-utils-5.8.1/debian/rules
--- xz-utils-5.6.0/debian/rules	2024-01-31 21:32:41
+++ xz-utils-5.8.1/debian/rules	2025-04-03 22:58:15
@@ -5,24 +5,23 @@
 	dh $@ $(opt_no_act)
 
 override_dh_auto_clean:
-	if [ -f doc/api.backup.done ]; then rm -rf doc/api; mv doc/api.backup doc/api; rm doc/api.backup.done; else rm -rf doc/api.backup*; fi
 	dh_auto_clean --builddirectory debian/xzdec-build
 	dh_auto_clean --builddirectory debian/normal-build
 	dh_auto_clean --builddirectory debian/static-build
 
 override_dh_auto_install-indep:
+	dh_auto_install --builddirectory debian/normal-build
 
 override_dh_auto_install-arch:
 	dh_auto_install --builddirectory debian/xzdec-build
-	dh_auto_install --builddirectory debian/normal-build
 	dh_auto_install --builddirectory debian/static-build
+	dh_auto_install --builddirectory debian/normal-build
 
 override_dh_installchangelogs:
 	dh_installchangelogs ChangeLog
 
 override_dh_auto_build-indep:
-	if [ ! -d doc/api.backup ]; then cp --reflink=auto -ra doc/api doc/api.backup; touch doc/api.backup.done; fi
-	cd doxygen && sh update-doxygen
+	dh_auto_install --builddirectory debian/normal-build
 
 override_dh_auto_build-arch:
 	dh_auto_build --builddirectory debian/normal-build
@@ -43,10 +42,22 @@
 	dh_auto_test --builddirectory debian/xzdec-build
 endif
 
-override_dh_auto_configure:
+override_dh_auto_configure-indep:
+ifneq (,$(findstring nocheck,$(DEB_BUILD_OPTIONS)))
+	@echo Skipping docs
+else
 	dh_auto_configure --builddirectory debian/normal-build -- \
 		$(opt_reproduce) \
 		--enable-threads --disable-static \
+		$(opt_optimize) $(opt_quiet) 	\
+		--enable-doxygen		\
+		--disable-xzdec --disable-lzmadec
+endif
+
+override_dh_auto_configure-arch:
+	dh_auto_configure --builddirectory debian/normal-build -- \
+		$(opt_reproduce) \
+		--enable-threads --disable-static \
 		$(opt_optimize) $(opt_quiet) \
 		--disable-xzdec --disable-lzmadec
 	dh_auto_configure --builddirectory debian/static-build -- \
@@ -91,7 +102,7 @@
     opt_optimize_small += --disable-assembler
 endif
 
-ifneq (,$(findstring n,$(MAKEFLAGS)))
+ifneq (,$(findstring n,$(firstword -$(MAKEFLAGS))))
     opt_no_act = --no-act
 endif
 
diff -Naur xz-utils-5.6.0/debian/source.lintian-overrides xz-utils-5.8.1/debian/source.lintian-overrides
--- xz-utils-5.6.0/debian/source.lintian-overrides	2024-01-31 21:32:41
+++ xz-utils-5.8.1/debian/source.lintian-overrides	2025-04-03 22:58:15
@@ -1,8 +1,2 @@
-# Source is tests/bcj_test.c
-source-is-missing [tests/compress_prepared_bcj_sparc]
-# Source is tests/bcj_test.c
-source-is-missing [tests/compress_prepared_bcj_x86]
 # This is a compressed text file with bad magic for testing.
 source-contains-prebuilt-windows-binary [tests/files/bad-1-v1-magic-2.lz]
-# Doxygen generated, #1019980, #1017094
-source-is-missing [doc/api/*.html]
diff -Naur xz-utils-5.6.0/debian/symbols xz-utils-5.8.1/debian/symbols
--- xz-utils-5.6.0/debian/symbols	2024-02-26 07:47:42
+++ xz-utils-5.8.1/debian/symbols	2025-04-03 22:58:53
@@ -24,5 +24,5 @@
  (arch=linux-any)lzma_stream_encoder_mt@XZ_5.2.2 5.2.2
  (arch=linux-any)lzma_stream_encoder_mt_memusage@XZ_5.1.2alpha 5.4
  (arch=linux-any)lzma_stream_encoder_mt_memusage@XZ_5.2.2 5.2.2
- XZ_5.6.0@XZ_5.6.0 5.6.0
- lzma_mt_block_size@XZ_5.6.0 5.6.0
+ *@XZ_5.6.0 5.6.2
+ *@XZ_5.8 5.8.0
diff -Naur xz-utils-5.6.0/debian/tests/testsuite xz-utils-5.8.1/debian/tests/testsuite
--- xz-utils-5.6.0/debian/tests/testsuite	2024-01-31 21:32:41
+++ xz-utils-5.8.1/debian/tests/testsuite	2025-04-03 22:58:15
@@ -2,7 +2,7 @@
 
 # Tweak the Makefile to link against the installed liblzma for the tests
 sed 's@..top_builddir..src.liblzma.liblzma.la@-llzma@' -i tests/Makefile.am
-
+./autogen.sh
 ./configure
 
 exec make check
diff -Naur xz-utils-5.6.0/debian/upstream/signing-key.asc xz-utils-5.8.1/debian/upstream/signing-key.asc
--- xz-utils-5.6.0/debian/upstream/signing-key.asc	2024-01-31 21:32:41
+++ xz-utils-5.8.1/debian/upstream/signing-key.asc	2025-04-03 22:58:15
@@ -13,18 +13,18 @@
 MtbltZo0ysPMa6Xj5xFaYqWk/BI4iLb2Gs+ByGo/+a0Eq4XYBMOpitNniQARAQAB
 tCdMYXNzZSBDb2xsaW4gPGxhc3NlLmNvbGxpbkB0dWthYW5pLm9yZz6JAlEEEwEK
 ADsCGwMCHgECF4AECwkIBwMVCggFFgIDAQAWIQQ2kMJAzlG0Zw0wrRw47nV9aRhG
-IAUCYzcN9AUJGPjTYgAKCRA47nV9aRhGIIlfD/0cS3q+z1fGoSYAR8FX/am39S6v
-1T0yVyicEFbwNd7x7OCJ6T8qZYb72FjArP8hq/S/1JQOYn7qAvzRb+5WLMgxEOUs
-rqol6jvwXGeyIhA09lSe7uEfv+NehEKs8F/30qIEa8Fwc7D6t/WCesDMULFMf4G8
-6XMFVs4gZDS2P54d7rt4+mDn2ZZve2OGcR3k3qr0sgZWNlLOk05n15XtBIFIBJbg
-hKXU68ud9I46Ji1Xrnqrn0H/P0jybW3klhp9AzUOfhuoRyUi1PLHDsw7d+jRAHGm
-8NnOPOAqxitYcAyU8SLBcpL3k24kjq6C6UwpStHBzqwQTQIfoyCVefxvlYm+8zht
-u/CmVXfjB0W+UImCWIeSpj/YzaD3i+YWz8m9lBGuU2WFNO7SAdoenJcUdNiwRL7P
-1E0n8HGM0OsSOks9NOWZHn+Ka36ivsNXAAec6inp0BnDEzdNcfrtgFMNBkiEeBD4
-+8T/HRNtBpYBrlDLtmk2uF86wRhSb+J+pOZeUVIuRQiHg3ridz2U+n6Ld4jqNRS1
-5L+yOhx2yJQ/lyIoAF0caPB3QcubXEYvnBLlZHvpdB5XwYr2ROv4OP5bniAvzL4P
-45jPdrWd/BJUpBD7V3pac83nvcB+z4K91Hnru2bTNatFlmAhR+1rN9nK9fggmeud
-BtgxpnVJAJt5VwF9p7kCDQRMxDmSARAAv8XAp2PGA/G1KmCrVIzOBm1NPIuqGAYP
+IAUCZ364UgUJHMSQwAAKCRA47nV9aRhGIH6dD/46yxNWPrEQlVFd2XbOfBdYjVSh
+SOtnF/7nfdpxObjsy5eqsuNGGknkfJkqDvYXcvtVLwn+U3WEscCX3eZ/syM+bpMj
++rMUfczDR21z6UEG9nfR5Yx+7TOpPfXylr2W1OfAtjdMv5UxYKpmmaQogVH+EeA0
+zAE0gZ+n1h0CzAXjmSeNOmAhaMcR1kkA7jM4ZW3r7ovcUuHSR6XwYl2gCDRkHaFR
+WePNaOUXRHn/WQFQusUZ8FgZdidw3j4U13ajpoQdZonXk9lBrxu3nS2EzaTuPsEf
+R9f1xRlWn733sTAJ5jCOpDFWvmLMNowodC5FYd3SKIn6K80uPamcVnPrtcfGZXAI
+OgELAmbrDRz3uDfK1sty4pqI7We/0l0poqf0i863piA0a+awOMURTo9B/vjJvdh1
+Rf672fVWfnWRfAnjgIifCN4nBPVKRcY9XgmOG8ZqQgpOug7EERz4W8IZccjxx4gu
+jTMXoPUY1s+6c/CJlQ3gESlfPkz1UYKCFjEvApMOD99SG8AUJvQbGvPFkIK5euWS
+hOVLAmLKI3YNGCXUiPA4i9AEYnkDtAydohUKU/WclBWHWaeAZDccDivs8dPiF3Dd
+YrR0eb81lB36Jm13cJodBS4aArHvgDeDvz7aLr7An0qXmNTnpc3cdU1ovskNXmA2
+Yj+I0ojpJsh6ici6dLkCDQRMxDmSARAAv8XAp2PGA/G1KmCrVIzOBm1NPIuqGAYP
 c1l9p0dYdhEgvfw0NXcl5MDv1jbOPZ2PspA8NP7Rqp6LNNXYTeM/eIJDndU5Phyi
 ewFpACAp7Gmm2dL5PUOhu0gIUnQYbN/QdGPoo7bNI646K1Y9aVTBu9fszQssjb6G
 qXHSNM+pskVn9lropO1tLrF0I9VSlSphlCmiQRlzBCZSnxD6UagkPaw1gJnJqnrd
@@ -35,65 +35,18 @@
 CjdACbWJsPEoIOrRFxY+NwJEA39Dkyalyh2l0qTNXTIYhLiDuzl+tWuBX+SjHavj
 9jGyvwr3T37gfzYCNMoZf8GaxAUJMCoGTqnsjTPGMion/DfdNkFDQ+fivdYiVQ9p
 /Njpr38sC83V8dHF/1KkIHImyzMPTdC7l/lMHyC2Gx2dWZOjuOOKit0Qoy3DZoQw
-vN1ZZND9M1UAEQEAAYkCOwQYAQoAJgIbDBYhBDaQwkDOUbRnDTCtHDjudX1pGEYg
-BQJjNw3ZBQkY+NNHAAoJEDjudX1pGEYg1mcP+IeCk90IFZ3SLX+siaZQCQIzcD+L
-3U+rwuiMPGcxBft5DBWetZe9lmsH9Q4JHj0B4wl8CIzL5U/oCc+J2wiZqWBF7w67
-ojM6RruuJyRNxoWyskDiCSYtcoMis8Eq2kBYmuz4Hei4MYBTX0R/HF3dJzOtfNYG
-olLta4vXzMuobBatsWpvTw0U60kHNixKRw3w61vFyhU8aZkTSH6KCRhtpkyw50GM
-KoTn4UHpIAZJ9msFqkrl2L73REDWPS1xdS9TA/h8JjRHH3BmuY9zVHYPwkesFDN9
-eah3rmt3VVI6r0HwBZnDWfqfANzvxJdoJP+b5KiY4OayQI5Jz+rqztNRva0rCy/O
-y32kSo7MSf5bvDA5bFzI4rFjgBXk58gy8HNA7bHrKSH5uSFSPnsii/Zvuva1T2sB
-VzV4ueNCA837y7zrT91B18y5rlUgKoUI3qUurJPjMe3TxLW2tsHHLNdkfxqqONOC
-7B7+rlqs4F/Gze1pBv848W4Zen2aCIJVAsZy0O3b/5018s6eZah5TLakGpLXfTaw
-EH6d+sYVMWYkLjKnNjy5AUb3x3EGjOowTzfoqsyGIFt4siCFeCw5Ace36BfJBXO/
-OyR51TFnibHNnUP/4l8mHDEa/FkWfmwAYtsCZFtegi/R24eT8lbYXW+39LTevN5z
-zLfsHcstacg70sqZAg0EY6xfcQEQANnHJ+JlWDtq5v4mgEb6w8KEdTBslz5IfsFY
-49u7gcs1+TplTOL8lIFggYgiuoaQDrQ+7UJumWRUXhQbRgEAAT+PXWm5YwevJ3FN
-j0ehz+tg4Gr7QASZacbRUg2OkXd00aWuaWKQ5HOlb8UAfLwHWcPw/u+6U+ihz0WC
-Q6YUzycA/bhNRFSHwbxwZjSOqlXFOJCoCL0o/FOWdjmzuGGBbQ13eYgXhBKg+h8w
-+dXkNNhLUsnDpcH0rcNHhuwPthgXZN1ga+WT9sk270MHU26JaOmfMsiQ923Glw85
-FeCJYTrC/MWdwmcbm2pt1D//KVkCM4VsKe0ww7u97cldaMQzqYMS6qWaVoesXbVj
-6ZA3mLYF1Ep8yJAcH0qy95PPYS/hw71NuXOyb+gO9hHb0s5GMmXc8TeOo1JVsAGK
-HY4acwBL3fjlrm6fW/I/fHr+gDJntDSIrNmaXjueyXjR1CTjPdomn3JreWmHkA1A
-/e7PBC0NJmCQPL8kPIgv/PX0W0CADtC6BaTHgexLOZmcTkZAkiYDT9fFdg1yTY2p
-D1gn9FjxbdoPh2macUBgTViLmS62tWV+rMq6MGmLf8l9IzWGQl1xEnuus7Dr0ldW
-9upVbUSJfyCnwpoZXqDQw38bXl0lP8Axj39fAQUhg/u9mXfZDHsDruOiQ0HtsoJn
-FZ7GmlGtABEBAAG0HEppYSBUYW4gPGppYXQwMjE4QGdtYWlsLmNvbT6JAlQEEwEK
-AD4WIQQi1GXytMFzgDsgxt5Z/PIH/qf0RQUCY6xfcQIbAwUJCWYBgAULCQgHAgYV
-CgkICwIEFgIDAQIeAQIXgAAKCRBZ/PIH/qf0RcmSEAC+99MK7bDwVF5oC/3ni5uV
-w3pIUG7Jdxc941njpkGbot8wStuPrAbIvNeJTdYy4+/AOWJfn8S2SGP6O6P9eUAr
-sO1FRrwl/pHa7WgVoah3HreCPMSg5etJiha3UA3AAMucynkxPU/Xg8clwIpLKPap
-xU5opW9ZhHDZh6gp3y1zRZybY0/EKT6fNRU9Pxs7raQk3imdqVYDe6W2jNmJKipw
-x6XAjxzNhRjdOIEv4Wo5RHbjfxMWdYMadojftleSNNhIQhG6d6je+MtwkOZLCt3I
-vc2whtfJKc3yBreRmLqGCMhL9OHcqErPJam6P3J0EwAyeJhQFEVsqX+kfw+3NDbn
-xOjFWOIb+EGsRupfQ/xHIB3piKIcM4+M+qPWWc156WIjK0G0fMfUcHdrYSDlljuI
-bgMyDzYhT+pwv2Vo8eyViL3cuG9pRH4A8COF3vU8B7FfFCQS2xQ48/JsVMJWsbrv
-1EIt5XgzfY9n31wEZjR6Q+jrnchNoDAFDJkm8BWw/kDSBtE9iiT9z+oT6V8XxS48
-/JQ4GTQhjKIDgxFbupK7aAWQ4dPnImDJ3sy2/Mm/NWe3eRNAKxB00YAhnw1PHo9t
-aP2q1KalfPcf5dEkf/3G9eEve/barza4F8AQyTeHfUoURIajmnBqu2HB9C9Av/oy
-6Z0skQtfOKMBjS/D2fEQ5bkCDQRjrF9xARAA0KjdV/x4fJOWG+6xDiKci3GhyBjT
-omGAEKTD2b9iJUjQ96R/RZq0LLV80gkmXI2CNAZD7fuks/p3bHDN0lzwabXlsaV+
-u2Hlph811wEtHVIwnsmdfTNyNGI/Aj5nxcBWga7u/MYhJwSikMyIjjFN1zsaWA/I
-TbsZ3YDQzDNHaVCLTdpIuWheHqH6PsYvgLd9iD79dy3/Dl3Wy01YOmo6XcFCqtxP
-09y+FlNIL3eDhhaEEU1mJgUvSkyD5aDrb/Inj2aNY6Rw8p2EqZYM7Hq9Xzw9pvDG
-H6xIppaHcNnbelaIyiEQ3S3KfAwgrUNPgPsHG4/iAIqE+R/Fwvx0yQUdup2B9QSB
-0OvVGK3G3gSXcBmgKTcZJzS4RIBanU1BwztmK5aWZP3QjRwc+Z/EktXq1EgzSILG
-ajC5UZA/zb+ptPULobroeBEyYDkKdpSwNrW9XEwXy2t1Tw+Svo6k+fXaCbDPfWPU
-nG9KrhHBLEuXvvAfeRMljeo7+UxiGrHT6YrnsI9piyOioDnGerNV23B8kje0/C22
-WuN+gNBKfZtpm7n2m9gL8hXE6CNOLIiC5LwtIGPTyI6pkugsG6o8PORDC9wIvxNz
-ER17bKgOLrAj5axd6+kFxjZ+Qi2pxEOI9AnsW4QCdEBrxJ9wUtQb1zAeDtGZ5Knl
-at87D7vF2qPUiYUAEQEAAYkCPAQYAQoAJhYhBCLUZfK0wXOAOyDG3ln88gf+p/RF
-BQJjrF9xAhsMBQkJZgGAAAoJEFn88gf+p/RFbx0P/21ConhtxFcs9etDNsGIylEG
-IupHGWl5LafAVSuvK/skJsBuBngaopre1H/mhR8gJnCBImFVQx4V9S0MA64tVEEQ
-cCSxXHO1TGkBXMXa0fpmu9Mjbq3MVDDlT7GtMSXd5rM2BkXpCyq/nfXdEjxKtQCG
-/3mzQDwJ1UiLdFnks8+yhP9eCC1pLPbyJJpnEMKGNjmM028LyyOnjjn56CmCVfGN
-Hbii1KFQNpkt3pgt8wdNHWto6ghl3yVtGhQAnfa+Re7ZRQ+AjoYLQdoTIdBDd9ob
-lIZStxa4Dpc4P2nte27dIOABQTqBvFR4JBaIWbZz50SiV22vifcFcg44p88eL+GI
-1UOYZbM+wHZfMJuhQ8d9aPinngj4gV/7C3ci5HQf06peKkv/3ITZweSnvNohW5zj
-fE9rYVU6QkvRecUzru1IwpVooUDqbJ7AcOu14EPJlW3ggfxyV6OGzpzBY5/0M5xs
-9A7UL3i8xz4aVqGgIvQUZLWZT8K6aI0HXMI/Sjb0hRdC8uT+YzzD/Mvrmx+KVt+b
-uQuTWXxQ7L+lTGThT0qm/v1ntohTBJlVeSR9C4eWYYKbTLpAotNplAvry4NesylY
-7Y7YH604ImJsYXN5w9513vDvsYqqMwddxVfcd3bCdkAxPAgmyMcbio78t8DyPN7n
-46F3erOKU/vOLCZorkwq
-=akRR
+vN1ZZND9M1UAEQEAAYkCPAQYAQoAJgIbDBYhBDaQwkDOUbRnDTCtHDjudX1pGEYg
+BQJnfrhHBQkcxJC1AAoJEDjudX1pGEYgjaIQAIxs2u0oNqKiFK58d/lpSg+sWgwO
+3WG9cChhQvPnJdyAdXKlnw8/SSykqtuMv1/N29mYSO7K3sCISvoCTR2KqT546o+C
+2/CqSs3WEuX1SpPQGVPULiTxLLsZh0WhJrKchmV3NRACPxoueiqtFosJPA6u1sf2
+ilN89ArGxzteIuTpyUy5r98GHUK5svjVh4sz+KvTNlGFHN3nAoR7BhiDf6RgQ0Q3
+ukkGJqvtYSUV3RuVkgR+7FHDFsb6RRmykcm+YStz3P0YTMJjbLYewaQLzKZnLWnS
+N18NRrITwEUicBXkcOFjF18d7jXYsAzzX7l28LFi5ygWuOOvT5s7EzTmWty5+qre
+HHnjL4SCQY2rkb9AipUTqpoDhBz91QivTdQ/EsJlP6e08hwaa7heZqZWWy1eRr8R
+EzD2r3GfQ2EegPkwsidivDBlK4PgWnlEl06RbE1dXoamXfPJ1q0SFlkw3zMW+bqY
+P1jTxrJVAlXFf8qT3d5pXz871R1NPP/dyPJni844Bi42q8JHIZaKWZ6QQsP20TGM
+BrvbI8qMCM1gH58vcBWBHtpm2kGEjVdJyl75RgjCf1VHqPab6A0/+tABGlLf5rm9
+kxOmb2wDCW8xPno/A8psVWcYE7gERT6Bdqbkr3SZw92izGCA4m17O1MPf6vj/ngb
+RROvuEI11HLOmBJv
+=m5Nx
 -----END PGP PUBLIC KEY BLOCK-----
diff -Naur xz-utils-5.6.0/debian/watch xz-utils-5.8.1/debian/watch
--- xz-utils-5.6.0/debian/watch	2024-02-26 07:47:42
+++ xz-utils-5.8.1/debian/watch	2025-04-03 22:58:15
@@ -1,4 +1,4 @@
 version=4
 opts=dversionmangle=s/\+\d{8}$//,pgpsigurlmangle=s/$/.sig/ \
-https://xz.tukaani.org/xz-utils/	\
-https://github.com/tukaani-project/xz/releases/download/v[^/]+/xz-([\d\.]*(?:beta|alpha)?)\.tar\.xz
+https://tukaani.org/xz/		\
+https://github.com/tukaani-project/xz/releases/download/v[^/]+/xz-(5\.8\.[\d\.]*(?:beta|alpha)?)\.tar\.xz
